legion_airflow/legion_airflow/hooks/k8s_secret_hook.py

```diff
@@ -0,0 +1,45 @@
+#
+#    Copyright 2018 IQVIA. All Rights Reserved.
+#
+"""K8S Connections hook package."""
+import os
+import json
+
+from airflow import LoggingMixin
+from airflow.hooks.base_hook import BaseHook
+from airflow.models import Connection
+
+from legion.k8s import K8SSecretStorage
+
+
+class K8SSecretHook(BaseHook):
+    """Hook for looking connections/credentials in K8S first than Airflow."""\
+
+    LOG = LoggingMixin().log
+    CONNECTIONS_SECRET = 'airflow-connections'
+
+    @classmethod
+    def get_connection(cls, conn_id: str):
+        """
+        Return connection by connection id.
+
+        :param conn_id: connection id
+        :type conn_id: str
+        :return: :py:class:`airflow.models.Connection` -- connection
+        """
+        namespace = os.environ.get('NAMESPACE')
+        try:
+            secret = K8SSecretStorage.retrive(storage_name=cls.CONNECTIONS_SECRET, k8s_namespace=namespace)
+            if conn_id in secret.data:
+                conn_data = secret.data.get(conn_id)
+                if conn_data:
+                    conn = json.loads(conn_data)
```

Comment:
Also we should not save all connection data in JSON. We should use different keys for different data (e.g. `login=secret['login']`)